Flag to add session attributes on request

dnetoMOB opened the pull request:
Adds the `withSessionAttributes` property that allows a request to be initialized with the provided attributes object.

hoegertn commented on index.js:
```diff
@@ -449,6 +450,12 @@ module.exports = {
+					// adds values from withSessionAttributes to the session
+					if (currentItem.withSessionAttributes) {
+						var session = request.session.attributes;
+						for(var x in currentItem.withSessionAttributes) session[x] = currentItem.withSessionAttributes[x];
```
Additionally, please change the following:

- use expanded form of `for`
- check if session has own property x
- rename `x` to a meaningful name

dnetoMOB commented:
- For loop expand to multi-line expression.
- renamed loop variable to more descriptive name
- added check so that an attribute provided in `withSessionAttributes` should not be added to the session if it already exists.

hoegertn commented on index.js:
```diff
@@ -449,6 +450,17 @@ module.exports = {
+						var session = request.session.attributes;
+
+						for (var newAttribute in currentItem.withSessionAttributes) {
+							if (typeof session[`${newAttribute}`] === "undefined") {
```
Why not `if (!session[newAttribute]) {`

dnetoMOB replied:
Good suggestion. I've had issues before with `"undefined"` vs `undefined`, but that syntax should clear things up.

hoegertn commented on index.js:
```diff
@@ -449,6 +450,17 @@ module.exports = {
+
+						for (var newAttribute in currentItem.withSessionAttributes) {
+							if (typeof session[`${newAttribute}`] === "undefined") {
+								session[`${newAttribute}`] = currentItem.withSessionAttributes[`${newAttribute}`];
```
Why do you use template strings here?

dnetoMOB replied:
Template strings removed (it was an odd suggestion from a friend).
